src/OpenFeature.Contrib.ConfigCat/ConfigCatProvider.cs

```diff
@@ -0,0 +1,118 @@
+        {
+            return ResolveFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return ResolveFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ResolveFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            var user = context?.BuildUser();
+            var result = await Client.GetValueDetailsAsync(flagKey, defaultValue?.AsObject, user);
+            var returnValue = result.IsDefaultValue ? defaultValue : new Value(result.Value);
+            var details = new ResolutionDetails<Value>(flagKey, returnValue, ParseErrorType(result.ErrorMessage), errorMessage: result.ErrorMessage, variant: result.VariationId);
+            if(details.ErrorType == ErrorType.None)
+            {
+                return details;
+            }
+
+            throw new FeatureProviderException(details.ErrorType, details.ErrorMessage);
+        }
+
+        private async Task<ResolutionDetails<T>> ResolveFlag<T>(string flagKey, EvaluationContext context, T defaultValue)
+        {
+            var user = context?.BuildUser();
+            var result = await Client.GetValueDetailsAsync(flagKey, defaultValue, user);
+            var details = new ResolutionDetails<T>(flagKey, result.Value, ParseErrorType(result.ErrorMessage), errorMessage: result.ErrorMessage, variant: result.VariationId);
+            if(details.ErrorType == ErrorType.None)
+            {
+                return details;
+            }
+
+            throw new FeatureProviderException(details.ErrorType, details.ErrorMessage);
+        }
+
+        private static ErrorType ParseErrorType(string errorMessage)
+        {
+            if (string.IsNullOrEmpty(errorMessage))
+            {
+                return ErrorType.None;
+            }
+            if (errorMessage.Contains("Config JSON is not present"))
+            {
+                return ErrorType.ParseError;
+            }
+            if (errorMessage.Contains("the key was not found in config JSON"))
+            {
+                return ErrorType.FlagNotFound;
+            }
+            if (errorMessage.Contains("The type of a setting must match the type of the specified default value"))
+            {
+                return ErrorType.TypeMismatch;
+            }
+            return ErrorType.General;
+        }
+
+        /// <summary>
+        /// Disposes the <see cref="Client"/>.
+        /// </summary>
+        public void Dispose()

```

Comment:
After taking a closer look at the OpenFeature SDK architecture, specifically at [this PR](https://github.com/open-feature/dotnet-sdk/pull/158), it looks like making the provider class implement `IDisposable` is not the correct approach after all.

Also, a proper `Dispose` implementation would be tricky because of the way ConfigCat SDK manages client instances: 

> You can acquire singleton client instances for your SDK keys using the ConfigCatClient.Get(sdkKey: "<sdkKey>") static factory method. (However, please keep in mind that **subsequent calls to ConfigCatClient.Get() with the same SDK Key return a shared client instance, which was set up by the first call**.)

That is, if you create two provider instances with the same SDK key and then dispose one of them, the underlying client will be disposed in the other one as well. Probably we would need some kind of reference counting to make this work correctly.

So, in light of all this, I say we don't implement `IDisposable` for now. Let's address this issue when the provider shutdown feature lands.

Reply:
Yeah, I wasn't aware of the Shutdown implementation. I'll remove the disposable implementation.